symbology/custom-dictionary-style/src/main/java/com/esri/samples/custom_dictionary_style/CustomDictionaryStyleSample.java

```diff
@@ -67,31 +91,124 @@ public void start(Stage stage) {
       // set the initial viewpoint to the Esri Redlands campus
       map.setInitialViewpoint(new Viewpoint(new Point(-1.304630524635E7, 4036698.1412000023, map.getSpatialReference()), 5000));
 
-      // create the restaurants feature table from the feature service
-      ServiceFeatureTable restaurantFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0");
-
-      // create the restaurants layer and add it to the map
-      FeatureLayer restaurantLayer = new FeatureLayer(restaurantFeatureTable);
-      map.getOperationalLayers().add(restaurantLayer);
-
-      // open the custom style file
-      String styleFilePath = new File(System.getProperty("data.dir"), "./samples-data/stylx/Restaurant.stylx").getAbsolutePath();
-      DictionarySymbolStyle restaurantStyle = DictionarySymbolStyle.createFromFile(styleFilePath);
-
-      // create a dictionary renderer
-      DictionaryRenderer dictionaryRenderer = new DictionaryRenderer(restaurantStyle);
-
-      // set the dictionary renderer to the restaurant layer
-      restaurantLayer.setRenderer(dictionaryRenderer);
-
-      // add the map view and control panel to stack pane
-      stackPane.getChildren().add(mapView);
+      // set up the UI
+      setupUI();
+
+      // enable UI interactions once the feature layer has loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          controlsVBox.setDisable(false);
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // create a portal item using the portal and the item id of the dictionary web style
+      Portal portal = new Portal("https://arcgisruntime.maps.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "adee951477014ec68d7cf0ea0579c800");
+
+      // create a stylex file from a local location
+      File stylxFile = new File(System.getProperty("data.dir"), "./samples-data/stylx/Restaurant.stylx");
+
+      // create a service feature table
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/Redlands_Restaurants/FeatureServer/0");
```

Comment:
I'm keen to get us thinking about using type inference more often to improve readability. this is a good example: we can use var and also split this over two lines and it becomes very readable:

```suggestion
      var serviceFeatureTable = 
      new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/Redlands_Restaurants/FeatureServer/0");
```